clients/upload.go

```diff
@@ -0,0 +1,64 @@
+package clients
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"net/url"
+
+	"github.com/livepeer/catalyst-api/log"
+	"github.com/livepeer/catalyst-api/video"
+)
+
+type SourceUpload struct {
+	S3    S3
+	Probe video.Prober
+}
+
+func (s *SourceUpload) Upload(args TranscodeJobArgs, s3HTTPTransferURL *url.URL) (TranscodeJobArgs, error) {
```

Comment:
Can we call this something more explicit like `CopySrcToDst` or something to that effect so that it doesn't clash with other functions like `UploadOSURL` etc?

Reply:
makes sense 👍